base/applications/kbswitch/kbswitch.c

```diff
@@ -436,6 +436,7 @@ WndProc(HWND hwnd, UINT Message, WPARAM wParam, LPARAM lParam)
 
         case WM_LANG_CHANGED:
         {
+            ulCurrentLayoutNum = GetNextLayout();
```

Comment:
My test show that it is only needed here, but i may missed something. But for sure it is not needed on every call. Because e.g., if you change it with mouse and gui, the correct item is selected already.